### ISSUE TITLE
Chapter, section, subsection heading numbers are wrong on EPUB output

### ISSUE THREAD

**Issue by N0rbert** [jgm/pandoc#5699]
I use latest pandoc 2.7.3-1 on Ubuntu MATE 16.04.6 LTS.

Below is the example markdown document with numbered headings:

```
# Header 1 with number {#intro}
# Header 1 without number try 1 {-#h1t1}
## Header 2 with number try 1 {#h2t1}
### Header 3 with number try 1 {#h3t1}

# Header 1 without number try 2 {-#h1t2}
## Header 2 with number try 2 {#h2t2}
### Header 3 with number try 2 {#h3t2}
```

The conversion process to HTML with `pandoc epub_heads.md -o out.html --number-sections` gives correct result:

```html
<h1 id="intro"><span class="header-section-number">1</span> Header 1 with number</h1>
<h1 id="h1t1" class="unnumbered">Header 1 without number try 1</h1>
<h2 id="h2t1"><span class="header-section-number">1.1</span> Header 2 with number try 1</h2>
<h3 id="h3t1"><span class="header-section-number">1.1.1</span> Header 3 with number try 1</h3>
<h1 id="h1t2" class="unnumbered">Header 1 without number try 2</h1>
<h2 id="h2t2"><span class="header-section-number">1.2</span> Header 2 with number try 2</h2>
<h3 id="h3t2"><span class="header-section-number">1.2.1</span> Header 3 with number try 2</h3>
```

But the conversion to EPUB with `pandoc epub_heads.md -o out.epub --number-sections` has wrong rendering (while bookmarks look correct):

![epub](https://user-images.githubusercontent.com/309456/63549202-9b8c1480-c538-11e9-8a92-8d605db016b8.png)

```html
...
<body epub:type="bodymatter">
<section id="h1t1" class="level1 unnumbered">
<h1>Header 1 without number try 1</h1>
<section id="h2t1" class="level2">
<h2><span class="header-section-number">0.1</span> Header 2 with number try 1</h2>
<section id="h3t1" class="level3">
<h3><span class="header-section-number">0.1.1</span> Header 3 with number try 1</h3>
</section>
</section>
</section>
</body>

...
```

(note that numbers start from zeros, that is incorrect)

**Comment by jgm** [comment]
You've marked the top-level headings "unnumbered" in chapters 2 and 3.
It's really not clear how the subordinate sections should be numbered in this case.
Generally, if you have an unnumbered section, the subsections should also be unnumbered.

Try this: use a numbered section heading (not `{-}`), and suppress display of the number using CSS. Something like

```css
h1 > span.heading-section-number { display: none }
```

**Comment by N0rbert** [comment]
Let me clarify why I'm reporting this.
I'm using `pandoc` indirectly from `bookdown` inside RStudio. The `bookdown` allows to have numbered and unnumbered headers (see [*2.1.2 Block-level elements*](https://bookdown.org/yihui/bookdown/markdown-syntax.html)).
The YAML-header to be added is the following:

```
---
output: 
  bookdown::pdf_document2: default
  bookdown::html_document2: default
  bookdown::epub_book: default
---
```

The code in the this bug-report leads to correct rendering/`knit`ing in all other formats, but fails in EPUB. See PDF as example:

![pdfout](https://user-images.githubusercontent.com/309456/64419573-dd26ce80-d0a5-11e9-9229-cf39bd15f1da.png)

That is why I simplified this issue to show that the problem is with `pandoc` itself. 
On the left side of [e-book reader screenshot](https://user-images.githubusercontent.com/309456/63549202-9b8c1480-c538-11e9-8a92-8d605db016b8.png) you can see that table of contents has correctly numbered headings, but heading numbering in document is wrong.
So I suppose that there is a bug in EPUB writer.

**Comment by jgm** [comment]
But it's not clear to me why the numbering you get in the LaTeX example and in the EPUB's navigation is "right."  The numbering treats the material under the level 1 heading as, essentially, part of the previous numbered section.  But the level one heading *should* start a new section.  And then it's not clear how to number.

Consider this kind of case:

```
# Chapter one

## Section a
## Section b

# Appendix {-}

## Appendix section a
## Appendix section b
```

Clearly it wouldn't make sense to number the appendix's subsections 1.3 and 1.4! They are part of the appendix, not section 1.

I think you're trying to use unnumbered headings here for a purpose they weren't designed for.  (I'm not exactly sure what you are trying to achieve -- maybe you could say more about that.)

**Comment by N0rbert** [comment]
I tried to use this semantics to realize the following structure:

```
# Part 1 heading {#part1-id}

. . .

# Lecture 1 {-}

. . .

## Lecture 1 topic 1 {#lec1-t1}

. . .

### Lecture 1 topic 1 subtopic 1 {#lect1-t1-st1}

# Lecture 2 {-}

. . .

# Part 2 heading {#part2-id}

# Lecture 3 {-}

and so on

```

Headings with "Lecture" should not be numbered in such my case. They are separators, but I need to have them with outstanding header. Adding `##` to *Lecture* will zero second number (that is `x.0.y`).

So the EPUB output 

![epub](https://user-images.githubusercontent.com/309456/64455589-bc3a9980-d0f6-11e9-8350-d711a81585ec.png)

differs from PDF and HTML:

![pdf-html](https://user-images.githubusercontent.com/309456/64455531-931a0900-d0f6-11e9-9c92-45e67cf7934b.png)

(got these files with simple `pandoc l.md -o l.epub --number-sections`, `pandoc l.md -o l.pdf --number-sections`, `pandoc l.md -o l.html --number-sections` without `bookdown`)

I expected that EPUB should be the same as HTML and PDF, that is why I reported this :)

**Comment by N0rbert** [comment]
Seems to be fixed with latest Pandoc 2.10.1 and latest bookdown 0.20.

**Comment by N0rbert** [comment]
Seems to be fixed with latest Pandoc 2.10.1 and latest bookdown 0.20.

**Comment by jgm** [comment]
yes, closing!
